python/cudf/cudf/core/dataframe.py

```diff
@@ -423,10 +470,66 @@ def _getitem_tuple_arg(self, arg):
 
     @_cudf_nvtx_annotate
     def _setitem_tuple_arg(self, key, value):
-        # TODO: Determine if this usage is prevalent enough to expose this
-        # selection logic at a higher level than ColumnAccessor.
-        for col in self._frame._data.get_labels_by_index(key[1]):
-            self._frame[col].iloc[key[0]] = value
+        columns_df = self._frame._from_data(
+            self._frame._data.select_by_index(key[1]), self._frame._index
+        )
+
+        error_msg = (
```

Comment:
Can we call this `shape_mismatch_error_msg` instead?

Reply:
Can we use positions rather than variables in this particular case for the string formatting? I don't think the variable names `value1` and `value2` add much to the readability.

So:

```python
shape_mismatch_error_msg = "shape mismatch: ... {0} ... {1}"

# later
raise ValueError(shape_mismatch_error_msg.format(x, y))
```

Reply:
Given how many lines in this function are dedicated to just raising this error message multiple times, I'd go so far as to suggest we should probably just make a helper to raise the value error:

```python
def shape_mismatch_error(x, y):
    raise ValueError(
        f"shape mismatch: value array of shape {x} "
        f"could not be broadcast to indexing result of "
        f"shape {y}"
    )
```